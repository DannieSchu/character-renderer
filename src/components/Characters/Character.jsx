import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, url }) => {
  return (
    <section>
      <h3>{name}</h3>
      <img src={url} alt={name} />
    </section>
  );
};

Character.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Character;
