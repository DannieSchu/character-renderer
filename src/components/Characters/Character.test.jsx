import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character.jsx';

describe('Character', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <Character name="Helga G. Pataki" url="https://vignette.wikia.nocookie.net/heyarnold/images/e/e5/Helga_Pataki.png/revision/latest/scale-to-width-down/266?cb=20190816021807" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
