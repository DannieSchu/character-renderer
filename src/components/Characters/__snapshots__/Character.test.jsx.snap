// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character matches a snapshot 1`] = `
<section>
  <h3>
    Helga G. Pataki
  </h3>
  <img
    alt="Helga G. Pataki"
    src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e5/Helga_Pataki.png/revision/latest/scale-to-width-down/266?cb=20190816021807"
  />
</section>
`;
